pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -107,10 +124,29 @@ func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageClust
 			},
 		},
 	}
-
+	// OBC StorageClass will be created only if it is not a cloud platform
+	if platform, err := r.platform.GetPlatform(r.client); err == nil &&
+		!r.isCloudPlatform(platform) {
+		ret = append(ret, r.newOBCStorageClass(initData))
+	}
 	return ret, nil
 }
 
+// isCloudPlatform returns true if the platform is cloud type
+// requires an explicit argument of type 'CloudPlatformType',
+// so that getting the platform and dealing with errors are
+// done outside of this function
+func (r *ReconcileStorageCluster) isCloudPlatform(platform CloudPlatformType) bool {
+	cloudPlatfroms := map[CloudPlatformType]struct{}{
```

Comment:
In this map way, it will be easier to add or remove a platforms from the cloud-platform (user or developer don't have to write another if condition or delete a block of code to remove it).

> This way you don't have to maintain a separate list here.
but we are any way maintaining a list either through map or directly in the code (through a sequence of if conditions)

But if you strongly oppose or prefer the condition way more, will change it.

Reply:
Please ignore the above comment, just now only I saw the `ValidCloudPlatforms`.. will change the code...

Reply:
Changed the function accordingly...